components/editor/utils/OverlayFormManager.js

```diff
@@ -58,9 +63,22 @@ class OverlayFormManager extends Component {
           {children({data: node.data, onChange})}
         </OverlayForm>
       )}
-      {preview}
+      {component || preview}
     </div>
   }
 }
 
+OverlayFormManager.defaultProps = {
+  showEditButton: true
+}
+
+OverlayFormManager.propTypes = {
+  showEditButton: PropTypes.bool.isRequired,
+  onChange: PropTypes.func.isRequired,
+  children: PropTypes.func.isRequired,
+  component: PropTypes.node,
+  preview: PropTypes.node,
+  extra: PropTypes.node
+}
+
```

Comment:
Props `node`, `editor` and  `attributes` should definitely get added here, because the component will never work without those.